### PR TITLE
Fixes #20272: Dockerify api-doc test, build and publication

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/20272

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: Jenkinsfile [hunk 1]
```diff
@@ -37,8 +37,29 @@ pipeline {
                         }
                     }
                 }
+                stage('typos') {
+                    agent { dockerfile { filename 'ci/typos.Dockerfile'
+                                         additionalBuildArgs  '--build-arg VERSION=1.0' } }
```
Comment (Fdall) [inline Jenkinsfile:42]: For consistency can you keep the closing brackets on a new line

File: Jenkinsfile [hunk 2]
```diff
@@ -104,11 +113,13 @@ pipeline {
...
                 stage('rudder-pkg') {
-                    agent { label 'script' }
+                    agent { docker { image 'python'
+                                     args '-v /etc/passwd:/etc/passwd:ro --tmpfs /srv/jenkins/.local:exec' } }
+
                     steps {
                         dir ('relay/sources') {
-                            sh script: 'typos --exclude "api-doc/*" --exclude "relayd/*"', label: 'check typos'
-                            sh script: 'make check', label: 'rudder-pkg tests'
+                            sh script: 'pip install -r rudder-pkg/requirements.txt', label: 'install dependencies'
```
Comment (Fdall) [inline Jenkinsfile:121]: Shouldn't the dependencies be installed at image build time?

Comment (amousset) [reply Jenkinsfile:121]: May be better indeed.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Review by Fdall — APPROVED** [review]

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**